explainer.md

```diff
@@ -132,45 +132,41 @@ MyScript also provides helper classes and SDKs to manage stroke capture. Applica
 
 ## Proposed Usage Example
 
-### Query Feature Support
+### Query Feature Support (V2)
 
 Handwriting recognizers on different platforms have different features. Web applications can query their feature support and decide if the API is suitable for their use case.
 
 ```JavaScript
-// The list of features to detect.
-await navigator.queryHandwritingRecognizerSupport({
-  'languages': ['en', 'zh-CN'],  // A list of languages
-  'alternatives': true,          // Can be any value
-  'unsupportedFeature': true,    // Can be any value
-})
+// Model constraints describes the recognizer's essential capabilities.
+const modelConstraints = {
+  // All languages to be recognized, in order of precedence.
+  languages: ['zh-CN', 'en'],
+}
+
+// Query which features are supported. This method takes model constraints
+// and returns a feature description.
+await navigator.queryHandwritingRecognizer(modelConstraints)
 
-// For each query:
-//   - If the recognizer supports its feature, returns true.
-//   - If the recognizer supports its feature, but the provided
-//     parameters aren't supported, return false.
-//     For example when proving a wrong language tag.
-//   - If the recognizer doesn't support its feature, the feature
-//     name is not included in the return value.
-//
 // => {
-//   languages: true,  // The recognizer supports both en and zh-CN
-//   alternatives: true,
-//   // Unsupported features are not included in the return value
+//   textAlternatives: true,
+//   textSegmentation: true,
+//   hints: {
+//     graphemeSet: true,
+//     textContext: true,,
+//     inputTypes: ['mouse', 'touch', 'stylus']
+//   }
 // }
+//
+// => `null` if the constraint can't be satisfied.
```

Comment:
I think conceptually they are different:

* `null` -> no such recognizer exists
* `{}` -> a recognizer exists, but doesn't support anything / browser choose not to disclose its capability

Also, `null` is easier to check, since `{}` is truthy.

Reply:
OK SG